File: dev/test_data/golang/test_main/aas_core_meta.v3/input/snippets/Verification/ID_shorts_of_variables_are_unique.go
// Check that [aastypes.IReferable.IDShort]'s among all the `inputVariables`,
// `outputVariables` and `inoutputVariables` are unique.
func IDShortsOfVariablesAreUnique[O1 aastypes.IOperationVariable,
		O2 aastypes.IOperationVariable,
		O3 aastypes.IOperationVariable] (
	inputVariables []O1,
	outputVariables []O2,
	inoutputVariables []O3) bool {
	idShortSet := make(map[string]struct{})

	var has bool

	if inputVariables != nil {
		for _, variable := range inputVariables {
			idShort := variable.Value().IDShort()

			if idShort != nil {
				_, has = idShortSet[*idShort]
				if has {
					return true
				}

				idShortSet[*idShort] = struct{}{}
			}
		}
	}

	if outputVariables != nil {
		for _, variable := range outputVariables {
			idShort := variable.Value().IDShort()

			if idShort != nil {
				_, has = idShortSet[*idShort]
				if has {
					return true
				}

				idShortSet[*idShort] = struct{}{}
			}
		}
	}

	if inoutputVariables != nil {
		for _, variable := range inoutputVariables {
			idShort := variable.Value().IDShort()

			if idShort != nil {
				_, has = idShortSet[*idShort]
				if has {
					return true
				}

				idShortSet[*idShort] = struct{}{}
			}
		}
	}

	return true
}

File: dev/test_data/csharp/test_main/aas_core_meta.v3/input/snippets/Verification/ID_shorts_of_variables_are_unique.cs
/// <summary>
/// Check that all <see cref="Aas.IReferable.IdShort" />'s are among all the
/// <paramref name="inputVariables" />, <paramref name="outputVariables" /> and
/// <paramref name="inoutputVariables" /> are unique.
/// </summary>
public static bool IdShortsOfVariablesAreUnique(
    IEnumerable<Aas.IOperationVariable>? inputVariables,
    IEnumerable<Aas.IOperationVariable>? outputVariables,
    IEnumerable<Aas.IOperationVariable>? inoutputVariables
)
{
    var idShortSet = new HashSet<string>();
    
    if (inputVariables != null)
    {
        foreach (var variable in inputVariables)
        {
            if (variable.Value.IdShort != null)
            {
                if (idShortSet.Contains(variable.Value.IdShort))
                {
                    return false;
                }
                idShortSet.Add(variable.Value.IdShort);
            }
        }
    }

    if (outputVariables != null)
    {
        foreach (var variable in outputVariables)
        {
            if (variable.Value.IdShort != null)
            {
                if (idShortSet.Contains(variable.Value.IdShort))
                {
                    return false;
                }
                idShortSet.Add(variable.Value.IdShort);
            }
        }
    }
    
    if (inoutputVariables != null)
    {
        foreach (var variable in inoutputVariables)
        {
            if (variable.Value.IdShort != null)
            {
                if (idShortSet.Contains(variable.Value.IdShort))
                {
                    return false;
                }
                idShortSet.Add(variable.Value.IdShort);
            }
        }
    }

    return true;
}
